src/engine/strat_engine/engine.rs

```diff
@@ -152,73 +145,43 @@ impl Engine for StratEngine {
                 // ref. https://github.com/stratis-storage/stratisd/issues/748
                 warn!("udev add: pool {} is already known, ignoring device {:?}!",
                       pool_uuid,
                       dev_node);
-                Ok(None)
+                None
             } else {
-
                 let mut devices = self.incomplete_pools
                     .remove(&pool_uuid)
                     .or_else(|| Some(HashMap::new()))
                     .expect("We just retrieved or created a HashMap");
                 devices.insert(device, dev_node);
 
-
-                let rc = match StratPool::setup(&dm, pool_uuid, &devices) {
+                match setup_pool(&DM::new()?, pool_uuid, &devices, &self.pools) {
                     Ok((pool_name, pool)) => {
-                        // We know we have a unique uuid, but the pools table requires a unique
-                        // name too so we will ensure unique before we insert as we don't want to
-                        // change the existing state if we have a conflict.
-                        //
-                        // We will also _not_ exit the daemon as we do in initialize as we were
-                        // previously up and running for some duration of time.
-                        if !self.pools.contains_name(&pool_name) {
-                            self.pools.insert(pool_name, pool_uuid, pool);
-                            Some(pool_uuid)
-                        } else {
-                            let dev_paths = devices
-                                .values()
-                                .map(|p| p.to_str().expect("Unix is utf-8"))
-                                .collect::<Vec<&str>>()
-                                .join(", ");
-
-                            self.incomplete_pools.insert(pool_uuid, devices);
-
-                            error!("udev add: duplicate pool name {:?} for uuid {:?}, \
-                                devices[{}], failing to setup complete pool.",
-                                   pool_name,
-                                   pool_uuid,
-                                   dev_paths);
-                            if let Err(e) = pool.teardown(&dm) {
-                                error!("Error while tearing down pool with duplicate name {:?}.",
-                                       e);
-                            }
-                            None
-                        }
+                        self.pools.insert(pool_name, pool_uuid, pool);
+                        Some(pool_uuid)
                     }
-                    Err(e) => {
-                        warn!("udev add: pool {} failed to setup, reason {:?}.",
-                              pool_uuid,
-                              e);
+                    Err(err) => {
+                        warn!("udev add: pool failed to setup, reason {:?}", err);
```

Comment:
Again, the pool UUID is removed, less helpful IMHO.

Reply:
Pool UUID is in the error returned by ```setup_pool()```.

Reply:
No, not all, not yet: https://github.com/stratis-storage/stratisd/pull/837#issuecomment-374616448.

Reply:
> I plan to enhance the error messages returned by setup_pool() to be more like the log message currently printed out, assuming there are no strong objections to this.

I would prefer you defer making changes to the existing messages and include the changes with your future PR to enhance the error messages by `setup_pool`.  IMHO it would be better to have redundant information in the mean time instead of no information until the improvements can be made, thanks.